### DISCUSSION TITLE
Analogue input to digital conversion

### DISCUSSION THREAD

**Discussion by sergiosolorzano** [Rubberazer/JETGPIO#20] [Q&A] [answered]
Great library thank you! 

I am quite a newbie on iot and jetson, if you came across this use case, can you please share thoughts how to use  this lib for analogue input to digital conversion? 

I was looking at High-Precision AD/DA Expansion Board for Raspberry Pi 4B/Raspberry Pi 3B+/Raspberry Pi3B/2B/Raspberry Pi Zero/Zero W/Zero WH/Raspberry Pi Zero 2W/ Zero 2WH/Jetson Nano, Onboard ADS1256/ DAC8552 Chip https://amzn.eu/d/5eWIC7x

Thank you!

**Comment by Rubberazer** [comment, marked as answer]
Never worked with this one but is similar to other low cost ones that I came across before, the hardware is looking OK for the Nano or the Orin Nano/NX. The pins that this AD/DA Converter needs fit on both Jetsons, I would advise to double check the pinout on your side anyways to be 100% before buying. 

So basically you talk to this board using SPI and at least a gpio working as output to enable/disable the chip (CS pin), the one on the Jetson doesn't match the one on this board so you have to implement an extra pin as output for that (or use cables instead of stacking the board up the Nano), either way not a big deal.

On the software side I believe that you can do this one with my library, but you should keep in mind that my lib is only going to give you SPI communications and that's it, all the wrapping around  to handle the converter e.g. setting up SPI mode, sending a wake up, reset, setting up the sampling rate and all that is going to be on you.

If you have experience with C or C++ it can be done very quickly by replacing the functions that implement the SPI part on the software provided by Waveshare by the ones on my library, another way around it is by cracking the datasheet and work it out from scratch. 

To dig up a bit more I checked here: https://www.waveshare.com/wiki/High-Precision_AD/DA_Board#Documentation All the info plus code, datasheets and all the rest is here. For what I have seen is enough to work everything out.

Have fun

**Reply by sergiosolorzano** [reply]
That's really helpful @Rubberazer , thanks a bunch !
